src/js/ripple/amount.js

```diff
@@ -283,6 +284,7 @@ Amount.prototype.product_human = function(factor, opts) {
   //
   // See also Amount#ratio_human.
   if (factor._is_native) {
+    product = product.clone();
     product._set_value(product._value.dividedBy(Amount.bi_xns_unit));
```

Comment:
This could be `return product._value.dividedBy(Amount.bi_xns_unit)` so you could get rid of the clone and mutation. Might be good to do the same for the other one too.